src/constructs/iam/policies/dynamodb.ts

```diff
@@ -15,7 +15,10 @@ abstract class GuDynamoDBPolicy extends GuAllowPolicy {
     super(scope, id, {
       ...props,
       actions: props.actions.map((action) => `dynamodb:${action}`),
-      resources: [`arn:aws:dynamodb:${scope.region}:${scope.account}:table/${props.tableName}`],
+      resources: [
+        `arn:aws:dynamodb:${scope.region}:${scope.account}:table/${props.tableName}`,
+        `arn:aws:dynamodb:${scope.region}:${scope.account}:table/${props.tableName}/index/*`,
```

Comment:
Are we sure this will work for the actions people are using in practice? There aren't many actions that support indexes as a resource, just the basics like Scan:
https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazondynamodb.html

If people have added other action props, this might break there stacks if CloudFormation then rejects the index resource for that action?

Reply:
Good question. I've tested that the CloudFormation that gets generated will successfully apply by deploying https://github.com/guardian/crosswordv2/pull/38 to our CODE environment (CF updates successfully and the expected behaviour can be observed with regard to permissions). 

I think this is similar to the case where wildcard resources can be applied to sets of actions. See the `AmazonDynamoDBReadOnlyAccess` managed policy attached. 
<img width="1139" alt="Screenshot 2023-09-18 at 10 42 29" src="https://github.com/guardian/cdk/assets/953792/c80d7a25-86e5-415f-87a2-2794ca56160a">

The [documentation that I can find](https://docs.aws.amazon.com/service-authorization/latest/reference/reference_policies_actions-resources-contextkeys.html) is not explicit on this point, but does state:

> If you specify a resource type in a statement with an action that does not support that resource type, then the statement doesn't allow access.

So I believe we're safe from disrupting existing stacks here. I will add a comment describing this to the change as it does seem like a potential point of confusion.